### PR TITLE
Add test refactoring.

### LINKED ISSUE #235: Refatorar testes de backend
## WHAT
Precisamos refatorar os códigos dos testes automatizados da nova arquitetura

## WHY
Com a finalidade de melhorar a legibilidade do código, e de evitar a duplicidade desnecessária de alguns trechos.

## CHECKLIST
- [x] Usar as constantes da classe Request para pegar o status HTTP ao invés de usar o número diretamente (200, 400, 404 e etc)
- [x] Usar as constantes da classe Request para pegar o verbo HTTP ao invés de usar o nome diretamente (GET, POST e etc)
- [x] Criar uma constante em cada classe para centralizar a URL base
- [ ] Isolar os `headers` adicional que é colocado nas requisições PATCH e POST

## ADDITIONAL INFO
Antes:
```php
public function testGetAgentsShouldRetrieveAList(): void
{
      $response = $this->client->request('GET', '/api/v2/agents');
      $content = json_decode($response->getContent());

      $this->assertEquals(200, $response->getStatusCode());
      $this->assertIsArray($content);
}
```

Depois:

```php
private const BASE_URL = '/api/v2/agents';
    
public function testGetAgentsShouldRetrieveAList(): void
{
      $response = $this->client->request(Request::GET, self::BASE_URL);
      $content = json_decode($response->getContent());

      $this->assertEquals(Response::HTTP_OK, $response->getStatusCode());
      $this->assertIsArray($content);
}
```

### PR DESCRIPTION
**Pull request by Akellymourab**
Resolve #235

### CODE REVIEW

**Review by saracamilo — APPROVED** [review]

File: app/tests/functional/AgentApiControllerTest.php [hunk 1]
```diff
@@ -11,102 +11,102 @@
...
+
+        $this->assertEquals($data['name'], $content['name']);
+        $this->assertEquals($data['shortDescription'], $content['shortDescription']);
+        $this->assertEquals($data['terms'], $content['terms']);
+        $this->assertEquals($data['type'], $content['type']);
+    }
+
+    public function testDeleteAgentShouldReturnSuccess(): void
+    {
+        $response = $this->client->request('DELETE', self::BASE_URL.'/2');
```
Comment (alessandrofeitoza) [inline app/tests/functional/AgentApiControllerTest.php]: aqui dá pra usar o `Request::METHOD_DELETE`

File: app/tests/functional/SpaceApiControllerTest.php [hunk 2]
```diff
@@ -11,82 +11,82 @@
...
+                'latitude' => '45.4215',
+                'longitude' => '-75.6981',
+            ],
+            'public' => true,
+            'shortDescription' => 'Uma breve descrição do espaço.',
+            'longDescription' => 'Descrição longa e detalhada do espaço.',
+        ]);
+
+        $response = $this->client->request(Request::METHOD_POST, self::BASE_URL, [
+            'headers' => ['Content-Type' => 'application/json'],
```
Comment (alessandrofeitoza) [inline app/tests/functional/SpaceApiControllerTest.php]: Pensando agora com a mente, imaginei que dava pra isolar esse trecho em um outro lugar, talvez em no `AbstractTestCase::setUp()`

algo tipo:
```php
$this->client = HttpClient::create()->withOptions([
    'base_uri' => 'http://localhost',
    'headers' => [
        'Content-Type' => 'application/json',
    ],
]);
```

**Review by alessandrofeitoza — CHANGES_REQUESTED** [review]
Pronto, só alguns pontos.

- Lembra de rodar localmente os testes, precisamos que todos estejam passando. 
- Aqueles que tiverem quebrados, adiciona essa linha no inicio do método 
```php
        $this->markTestSkipped();
```

**Comment by Akellymourab** [comment]
Agent Api Controller (App\Tests\AgentApiController)
 ✔ Get agents should retrieve a list
 ✔ Get one agent should retrieve a object
 ✔ Get agent types should retrieve a list
 ↩ Get agent opportunities should retrieve a list
 ↩ Create agent should create an agent
 ↩ Delete agent should return success
 ↩ Update
 ✔ Update not founded resource

Event Api Controller (App\Tests\EventApiController)
 ✔ Get event types should retrieve a list
 ✔ Get one agent should retrieve a object
 ↩ Get events by spaces should retrieve a list
 ↩ Post event should create a new event
 ↩ Delete event should remove an event
 ↩ Update event should update an event
 ✔ Update not founded event resource

Opportunity Api Controller (App\Tests\OpportunityApiController)
 ✔ Get opportunities should retrieve a list
 ✔ Get one opportunity should retrieve a object

Project Api Controller (App\Tests\ProjectApiController)
 ✔ Get projects should retrieve a list
 ✔ Get one project should retrieve a object
 ✔ Create project should create a project
 ↩ Delete project should return success
 ✔ Update project should update a project
 ✔ Update not founded project resource

Seal Api Controller (App\Tests\SealApiController)
 ✔ Get seals should retrieve a list
 ✔ Get one seal should retrieve a object
 ↩ Create seal should return created seal
 ↩ Delete seal should return success
 ↩ Update
 ↩ Update not founded resource

Space Api Controller (App\Tests\SpaceApiController)
 ✔ Get spaces should retrieve a list
 ✔ Get one space should retrieve a object
 ↩ Create space should return created space
 ↩ Delete space should return success
 ↩ Update
 ✔ Update not founded resource

Welcome Api Controller (App\Tests\WelcomeApiController)
 ✔ Homepage from a p i
 ✔ Method d e l e t e from a p i
 ✔ Method p o s t from a p i
 ✔ Error response when method is not allowed

OK, but some tests were skipped!
Tests: 39, Assertions: 54, Skipped: 16.

**Review by alessandrofeitoza — APPROVED** [review]

**Pull request merged**